lib/Backend/Lower.cpp

```diff
@@ -28592,6 +28657,50 @@ Lowerer::InsertAndLegalize(IR::Instr * instr, IR::Instr* insertBeforeInstr)
     LowererMD::Legalize(instr);
 }
 
+void
+Lowerer::InsertObjectCheck(IR::RegOpnd * funcOpnd, IR::Instr* insertBeforeInstr, IR::Instr*bailOutIfNotObject)
```

Comment:
>InsertObjectCheck [](start = 9, length = 17)

nit: I would rather pass the bailout info and kind to these functions (`InsertObjectCheck, InsertFunctionTypeIdCheck, InsertFunctionInfoCheck`), create the bailout instr inside them, insert it in the IR, and return that instr.